extension-impl/bootstrap-grpc/src/main/java/com/dib/sofa/rpc/bootstrap/grpc/GrpcConsumerBootstrap.java

```diff
@@ -0,0 +1,300 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.grpc.BindableService;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.AbstractStub;
+
+import com.alipay.sofa.rpc.log.Logger;
+import com.alipay.sofa.rpc.log.LoggerFactory;
+import com.alipay.sofa.rpc.proxy.Proxy;
+
+import javassist.CannotCompileException;
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.CtConstructor;
+import javassist.CtField;
+import javassist.CtMethod;
+import javassist.Modifier;
+import javassist.NotFoundException;
+import javassist.LoaderClassPath;
+import javassist.util.proxy.MethodHandler;
+import javassist.util.proxy.ProxyFactory;
+import javassist.util.proxy.ProxyObject;
+import com.alipay.sofa.rpc.invoke.Invoker;
+import com.alipay.sofa.rpc.message.MessageBuilder;
+
+import com.alipay.sofa.rpc.core.exception.RpcErrorType;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.exception.SofaRpcRuntimeException;
+
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+
+/**
+ * Consumer bootstrap for grpc 
+ *
+ * @author <a href=mailto:yqluan@gmail.com>Yanqiang Oliver Luan (neokidd)</a>
+ */
+@Extension("grpc")
+public class GrpcConsumerBootstrap<T> extends DefaultConsumerBootstrap<T> {
+
+    /**
+     * proxy of grpc client's XXXBlockingStub
+     */
+    ProxyObject                 proxyObject;
+
+    /**
+     * Set by ConsumerConfig.setInterfaceID(), or interface attribution in XML service definition.
+     * 
+     * PLEASE NOTE: For GRPC transport, interfaceID won't refer to a real interface, it should be
+     * FULL QUALIFIED name of the concrete CLASS, which is generated from .proto file.  
+     * e.g. GreeterGrpc.class.getName()
+     */
+    private final String        interfaceID;
+
+    /**
+     * Name of the service in .proto file.
+     * e.g. Greeter 
+     */
+    private final String        serviceName;
+
+    /**
+     * full qualified name of the blockingStub.
+     * e.g. GreeterBlockingStub
+     */
+    private final String        blockingStubName;
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(GrpcConsumerBootstrap.class);
+
+    /**
+     * 构造函数
+     
+     * @param consumerConfig 服务消费者配置
+     */
+    protected GrpcConsumerBootstrap(ConsumerConfig<T> consumerConfig) {
+        super(consumerConfig);
+        this.interfaceID = consumerConfig.getInterfaceId();
+        String[] segments = this.interfaceID.split("\\.");
+        String lastSegment = segments[segments.length - 1];
```

Comment:
why not use method `lastIndexOf `

Reply:
Will make change to use method lastIndexOf